### ISSUE TITLE
Can we fine-tuning the Text-to-Code Retrieval task?

### ISSUE THREAD

**Issue by pdhung3012** [salesforce/CodeT5#146]
Hello 
I wonder if we can finetune the text-to-code retrieval task for Text-to-Code Retrieval like UniXcoder at [here](https://github.com/microsoft/CodeBERT/tree/master/UniXcoder/downstream-tasks/code-search).
I have run the zero-shot code retrieval for Javascript. It shows that the best accuracy I can get for code retrieval is 70.2%, which is lower than the fine-tuned CodeT5+ at 71.3\% (reported in CodeT5+ paper at [here](https://arxiv.org/pdf/2305.07922.pdf). So I want to check if I can increase the zero-shot result by fine-tuning.

Thank you

**Comment by yuewang-cuhk** [comment]
Yes, you can definitely finetune on labeled datasets using contrastive loss (or combined with the matching loss) to further boost the retrieval performance.  We plan to release the finetuning scripts in the future if there are many asks for this.

**Comment by gzt4se** [comment]
> Yes, you can definitely finetune on labeled datasets using contrastive loss (or combined with the matching loss) to further boost the retrieval performance. We plan to release the finetuning scripts in the future if there are many asks for this.

I would like to ask if there are now open-source finetune scripts to share for Text-to-Code Retrieval using codet5+, thanks a lot!
